src/Tests/CoreTests/Output/OutputBuilderTest.cs

```diff
@@ -70,5 +72,104 @@ public void TestReplace()
             
             Assert.AreEqual(correctedMessage, parsedMessage);
         }
+        
+        [Test]
+        public void TestReplaceWithNullNewStr()
+        {
+            var replaceWord = "big brown";
+            var correctedMessage = "The  dog.";
+            
+            var output = new OutputBuilder().Append(originalMessage);
+            output.Replace(replaceWord, null);
+            var parsedMessage = output.Parse(terminalOptions);
+            Assert.AreEqual( correctedMessage, parsedMessage );
+        }
+                
+        [Test]
+        public void TestReplaceWithNullOldStr()
+        {
+            var output = new OutputBuilder().Append(originalMessage);
+            output.Replace(null, null);
+            var parsedMessage = output.Parse(terminalOptions);
+            Assert.AreEqual( originalMessage, parsedMessage );
+        }
+
+        [Test]
+        public void TestAppend_IntOverload()
+        {
+            var testData = new List<Tuple<string, int>>()
+            {
+                new Tuple<string, int>( "Negative number: ", -100 ),
+                new Tuple<string, int>( "Test: ", 100 ),
+                new Tuple<string, int>( "Pretty long text: ", 10000 ),
+                new Tuple<string, int>( "Very long text with very long number: ", int.MaxValue - 1 ),
+                new Tuple<string, int>( "Edge positive: ", int.MaxValue ),
+                new Tuple<string, int>( "Edge negative: ", int.MinValue )
+            };
+            var expectedResults = new List<string>()
+            {
+                "Negative number: -100",
+                "Test: 100",
+                "Pretty long text: 10000",
+                "Very long text with very long number: 2147483646",
+                "Edge positive: 2147483647",
+                "Edge negative: -2147483648"
+            };
+            var actualResults = new List<string>();
+            
+            foreach ( var testCase in testData )
+                actualResults.Add( new OutputBuilder().Append( testCase.Item1 ).Append( testCase.Item2 ).Parse( terminalOptions ) );
+
+            CollectionAssert.AreEqual( expectedResults, actualResults );
+        }
+
+        [Test]
+        public void TestClear()
+        {
+            var output = new OutputBuilder().Append( originalMessage );
+            var outputBuilderType = output.GetType();
+            var bufferPos = outputBuilderType.GetField( "bufferPos", BindingFlags.Instance | BindingFlags.NonPublic );
```

Comment:
nit: Whoa, this is cool, but I feel it's reaching into private implementation details we can probably validate with the public interface.  E.G. a refactoring that simply renamed "bufferPos" in OutputBuilder should not fail any unit tests if the user-facing (public) results of the code were not regressed.

We could have a test that proves we can call Parse multiple times (E.G. calling Parse twice in a row produces the same resulting string both times).  Then this test could:
* expect output.Parse(...) to be originalMessage
* output.Clear()
* expect output.Parse(...) to be empty string (or null? whichever should be expected)

Reply:
But, as you said at prev comment, probably it will be better if we have separate test cases for some methods. I have experience, when i needed to cover with test some small part, that may cause really big problems.
I can write another test in combination with other methods. In some way it will be a duplication, but it will show how method works with another methods.